### DISCUSSION TITLE
Dynamically build TSN Network

### DISCUSSION THREAD

**Discussion by m5awi** [inet-framework/inet#944] [Showcases]
I would like to dynamically build TSN network without using the ned or ini files, similar to the examples in samples/routing/netbuilder.cc.
The problem is that I can't inherit the network from the TsnNetworkBase as shown below. It would show me an error that the network is empty. which is true, since my network started to be built after I ran the simulation.
I want to inherit from TsnNetworkBase since I want also to modify parameters that come from that class (e.g: Bitrate). Is there way to go around this?
 Is there an example that shows how to build TsnNetwork dynamically?
```
simple NetBuilder 
{
  parameters:
        @display("i=block/cogwheel_s");
        string topologyFile;
        string streamFile;
}

network Network extends TsnNetworkBase
{
        submodules:
        builder: NetBuilder {
            topologyFile = "sources/input/mytop.top";
            streamFile = "sources/input/mypat.pat";
        }
}

```

**Comment by rhornig** [comment]
The TsnNetworkBase contains only components that can configure an **existing** network. If the network is not existing, you cannot configure it.

I just don't understand this:
> I want to inherit from TsnNetworkBase since I want also to modify parameters that come from that class (e.g: Bitrate). 

how TsnNetworkBase is related to classes, or Bitrate?

as for building up a TSN network dynamically. It's exactly the same as with any other module. You just can't configure them automatically. You have to configure all the TSN parameters, schedules manually (as you can use the TsnNetworkBase)

**Reply by m5awi** [reply]
Thanks for answering, The problem bitrate turned out to be because I forgot to call the function `callInitialize()` to build the inner modules.  I thought that the parameter bitrate came from the TsnNetworkBase class.
